website/docs/r/nsxt_network_segment_profile.html.markdown

```diff
@@ -0,0 +1,152 @@
+
+```hcl
+data "vcd_nsxt_segment_profile_template" "complete" {
+  name = "complete-profile"
+}
+
+data "vcd_nsxt_edgegateway" "existing" {
+  org  = "my-org"
+  name = "my-gw"
+}
+
+resource "vcd_network_routed_v2" "net1" {
+  org  = "my-org"
+  name = "routed-net"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  gateway       = "1.1.1.1"
+  prefix_length = 24
+
+  static_ip_pool {
+    start_address = "1.1.1.10"
+    end_address   = "1.1.1.20"
+  }
+}
+
+resource "vcd_nsxt_network_segment_profile" "custom-prof" {
+  org            = "my-org"
+  org_network_id = vcd_network_routed_v2.net1.id
+
+  segment_profile_template_id = data.vcd_nsxt_segment_profile_template.complete.id
+}
+```
+
+## Example Usage (Custom Segment Profile assignment to Org VDC Network)
+
+```hcl
+data "vcd_nsxt_manager" "nsxt" {
+  name = "nsxManager1"
+}
+
+data "vcd_nsxt_segment_ip_discovery_profile" "first" {
+  name            = "ip-discovery-profile-0"
+  nsxt_manager_id = data.vcd_nsxt_manager.nsxt.id
+}
+
+data "vcd_nsxt_segment_mac_discovery_profile" "first" {
+  name            = "mac-discovery-profile-0"
+  nsxt_manager_id = data.vcd_nsxt_manager.nsxt.id
+}
+
+data "vcd_nsxt_segment_spoof_guard_profile" "first" {
+  name            = "spoof-guard-profile-0"
+  nsxt_manager_id = data.vcd_nsxt_manager.nsxt.id
+}
+
+data "vcd_nsxt_segment_qos_profile" "first" {
+  name            = "qos-profile-0"
+  nsxt_manager_id = data.vcd_nsxt_manager.nsxt.id
+}
+
+data "vcd_nsxt_segment_security_profile" "first" {
+  name            = "segment-security-profile-0"
+  nsxt_manager_id = data.vcd_nsxt_manager.nsxt.id
+}
+
+data "vcd_nsxt_edgegateway" "existing" {
+  org  = "my-org"
+  name = "nsxt-gw-v40"
+}
+
+resource "vcd_network_routed_v2" "net1" {
+  org  = "my-org"
+  name = "routed-net"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  gateway       = "1.1.1.1"
+  prefix_length = 24
+
+  static_ip_pool {
+    start_address = "1.1.1.10"
+    end_address   = "1.1.1.20"
+  }
+}
+
+resource "vcd_nsxt_network_segment_profile" "custom-prof" {
+  org            = "my-org"
+  org_network_id = vcd_network_routed_v2.net1.id
+
+  ip_discovery_profile_id     = data.vcd_nsxt_segment_ip_discovery_profile.first.id
+  mac_discovery_profile_id    = data.vcd_nsxt_segment_mac_discovery_profile.first.id
+  spoof_guard_profile_id      = data.vcd_nsxt_segment_spoof_guard_profile.first.id
+  qos_profile_id              = data.vcd_nsxt_segment_qos_profile.first.id
+  segment_security_profile_id = data.vcd_nsxt_segment_security_profile.first.id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional, but required if not set at provider level) Org name 
+* `org_network_id` - (Required) Org VDC Network ID
+* `segment_profile_template_id` - (Optional) Segment Profile Template ID to be applied for this Org
+  VDC Network
+* `ip_discovery_profile_id` - (Optional) IP Discovery Profile ID. Can be looked up using
```

Comment:
When I see "can be looked up" I expect something like `vcd_resource_list`, where I get a list of entities of the same type. Instead, here I need to supply a name to get a data source.
I think it would be more understandable to say: "Can be referenced using a `vcd_nsxt_segment_ip_discovery_profile` data source"
Same goes for the arguments below.